src/pages/GE_TokenizeCard.page

```diff
@@ -72,49 +25,85 @@
             if (event && event.origin !== lightningOrigin) {
                 // Reject any messages from an unexpected origin
                 return;
+            }
+            if (event.data) {
+                handleMessage(event);
+            }
+        }
+
+        function handleMessage(event) {
+            const action = event.data.action;
+            switch (action) {
+                case 'mount':
+                    handleMounting(determinePaymentMethod(event.data.paymentMethod));
+                    break;
+
+                case 'setPaymentMethod':
+                    handleSetPaymentMethod(event.data.paymentMethod);
+                    break;
 
-            } else if (event.data && event.data.action === 'createToken') {
-                if (event.data.nameOnCard) {
-                    nameOnCard = event.data.nameOnCard;
-                } else {
-                    nameOnCard = DEFAULT_NAME_ON_CARD;
-                }
+                case 'createToken':
+                    handleTokenCreation(event.data.nameOnCard || DEFAULT_NAME_ON_CARD);
+                    break;
 
-                handleTokenCreation();
+                case 'createAchToken':
+                    handleAchTokenCreation();
+                    break;
             }
         }
 
+        function determinePaymentMethod(paymentMethodValue) {
+            return paymentMethodValue === 'ACH'
+                ? sfdo.PaymentMethod.ACH
+                : sfdo.PaymentMethod.CREDIT_CARD;
+        }
+
+        function handleSetPaymentMethod(paymentMethod) {
+            sfdo.setPaymentMethod(determinePaymentMethod(event.data.paymentMethod))
```

Comment:
I think what you want to pass to `determinePaymentMethod` is `paymentMethod` not `event.data.paymentMethod`

Reply:
fixed! Thanks for catching that!